src/mom5/ocean_diag/ocean_adv_vel_diag.F90

```diff
@@ -840,71 +840,47 @@ subroutine transport_on_s(Time, Adv_vel)
   ! mass transports leaving faces of grid cells
   wrk1=0.0
 
-  if (id_tx_trans > 0) then 
+  if (id_tx_trans > 0 .or. id_tx_trans_int_z > 0) then 
+    wrk1_2d(:,:) = 0.0
     do k=1,nk
          do j=jsc,jec 
             do i=isc,iec
                wrk1(i,j,k) = Adv_vel%uhrho_et(i,j,k)*Grd%dyte(i,j)*transport_convert
+               wrk1_2d(i,j) = wrk1_2d(i,j) + wrk1(i,j,k)
             enddo
          enddo
```

Comment:
I had a suggestion here to suggest replacing the i-j loops with implicit vectorized loops.
I deleted it because the arrays are data-domain (`isd:ied`) rather than compute domain(`isc:iec`).

But noting here that it might still be worthwhile to do the vectorised loop, even if it does more work, although it will clobber halo content.